User Management Redesign and User Details Page

Jacobjeevan opened the pull request:
## Proposed Changes

- Fixes #8878
- Fixes #9001 
- Fixes #8983
- BE PR https://github.com/ohcnetwork/care/pull/2595

### To Do:

- Open an issue for moving Profile: I've already moved most of the elements from existing profile page to to user details page (when it's details of logged in user), but there's a few issues in the works modifying certain elements in the profile page, so may need to tweak user details page.
- Done: While testing, I noticed that user deletion doesn't work for facility users page (delete route sets `is_active flag` to false, which works fine for userList (`/api/v1/users/`), but not for getFacilityUsers (`facility/facilityId/get_users/`) as latter is checking for `deleted=False` flag. BE PR [here](https://github.com/ohcnetwork/care/pull/2638)


### Potential things to check:

- Conditional rendering of elements
   - Qualifications (under Professional Information) only for Doctors/Nurses
   - YOE and Medical registration number (under Professional Information) only for Doctors
   - Logged in user can edit their own profile as well as change password
   - Edit (other users) only available for District/StateAdmin and superusers only
   - Deletion only (other users) for District/StateAdmins and superusers only
       - Only can delete users below in hierarchy
   - Avatar edit option (for all users) only for super user
   - Logged in user can edit their own avatar
- Users without edit permissions can view profiles of other users


<img width="1222" alt="image" src="https://github.com/user-attachments/assets/2a6fb1f1-9140-4243-a530-07dcbb34da27">

<img width="1222" alt="image" src="https://github.com/user-attachments/assets/8f6c4fde-8da7-4815-bc3a-f339b50ac848">

<img width="1217" alt="image" src="https://github.com/user-attachments/assets/1eb5e58a-7358-4288-9ecc-317cfa1c3e94">

<img width="1210" alt="image" src="https://github.com/user-attachments/assets/59cfb792-bd72-4079-9286-f5c0f2d4463b">

<img width="1290" alt="image" src="https://github.com/user-attachments/assets/5e16f775-7aa9-474c-b200-299d88d048a6">

<img width="1285" alt="image" src="https://github.com/user-attachments/assets/4b87daec-30e4-4e79-9e81-6c4b2bcf58de">

<img width="1288" alt="image" src="https://github.com/user-attachments/assets/f6ba8790-5602-4fe6-8b74-d6255172a016">

<img width="1206" alt="image" src="https://github.com/user-attachments/assets/a9145c31-1208-44bf-a9b3-741f7154877b">

<img width="1208" alt="image" src="https://github.com/user-attachments/assets/764ad599-7ddc-4f30-bd08-93991983183d">

<img width="546" alt="image" src="https://github.com/user-attachments/assets/2964a810-f223-4ea2-a638-0f6aad1de287">

If user doesn't have edit permissions:

<img width="1294" alt="image" src="https://github.com/user-attachments/assets/bd512e8b-a323-4f97-818b-ca56f94f03fe">

<img width="1212" alt="image" src="https://github.com/user-attachments/assets/3b4fb7fe-00af-40c8-a893-fd1541374c5c">

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

- **New Features**
	- Enhanced user management functionalities with new localization entries for account actions.
	- Introduced components for managing linked facilities and skills, including confirmation dialogs.
	- Added user avatar management capabilities, allowing users to upload and delete profile pictures.
	- Implemented a comprehensive user editing interface with toggling between view and edit modes.
	- Added a new component for displaying linked skills associated with a user.
	- Introduced a new component for managing user summaries and interactions.
	- Added a new `LinkedSkillsTab` component for displaying linked skills associated with a user.
	- Introduced a `UserHome` component for user management with dynamic tab rendering.
	- Added a `UserBanner` component for displaying user information in a banner format.
	- Introduced a `UserResetPassword` component for managing password resets.
	- Enhanced error handling and validation logic across various user management components.

- **Bug Fixes**
	- Improved validation logic for user forms, ensuring accurate error messages and handling.
	- Streamlined state management in user-related components for better performance.

- **Documentation**
	- Updated internal documentation to reflect new props and components introduced in this release.

- **Tests**
	- Added new end-to-end tests for user homepage functionalities and user management scenarios.
	- Updated existing tests to reflect changes in error messages and selectors for user verification.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Jacobjeevan committed:
User page and components

- Creating a separate user detail page (for user management) Pt 1

Jacobjeevan committed:
Minor fixes/refactoring

Jacobjeevan committed:
Added LinkedFacilitiesTab and components

Jacobjeevan committed:
Adding Roles and LinkedSkills

Jacobjeevan committed:
removing unused fields in UserInformation

Jacobjeevan committed:
adjusted user detail page urls

Jacobjeevan committed:
Page title adjustments

Jacobjeevan committed:
minor tweaks

Jacobjeevan committed:
FormEdits, fixes, customization

- Remove role editing from role management component
- Use passable formData for formValidation
- Minor customizations for role management issues
- Only show role management options for pages of users with Roles: Doctor and Nurse

Jacobjeevan committed:
Editing userAdd component to be reusable

Jacobjeevan committed:
Convert addUser to be reusable Pt2

- Import the form into UserInformationTab (removing the current form)
- Moved out the form components into its own file
	- Import the form into UserAdd

Jacobjeevan committed:
form validation edits and translations

Jacobjeevan committed:
Merge branch 'develop' into issues/8878-user-management-redesign

Jacobjeevan committed:
Manage Users: card modifications

Jacobjeevan committed:
Added list and card views

Jacobjeevan committed:
Refactor

- Refactor FacilityUsers to use similar format as ManageUsers
- Extracted out the list/card logic to UserListAndCard
- Changed FacilityUsers to use userList route (with facilityId as home_facility parameter)

Jacobjeevan committed:
Route and perms updates

- Modified to use a new route instead
	- Added a new route to Backend rather than changing permissions for UserModel
- Display view details button based on permissions

Jacobjeevan committed:
Refactoring

- Moving out additional common functionality inyo UserListAndCard
	- Removed memoization since it was causing hook errors
- Switched back to using getFacilityUsers route in FacilityUsers as that should be faster

Jacobjeevan committed:
update import paths and minor fixes

- password reset form fix
	- use passed in username than authUser's username
- UserAddEditForm
	- pass in facility Ids than full facilities in handleSubmit

Jacobjeevan committed:
Delete unused modals, reuse older ones

Jacobjeevan committed:
Merge branch 'develop' into issues/8878-user-management-redesign

Jacobjeevan committed:
minor styling changes and reordering

Jacobjeevan committed:
Conditionally render delete button/password reset form

- Added error message notifications for the form submits

Jacobjeevan committed:
Reword role to qualifications

Jacobjeevan committed:
RabbitAi suggestions and form fixes

Jacobjeevan committed:
Merge branch 'develop' into issues/8878-user-management-redesign

Jacobjeevan committed:
show district column only if data is available

Jacobjeevan commented on src/components/Users/LinkedFacilitiesTab.tsx:
```diff
@@ -0,0 +1,33 @@
+import LinkedFacilities from "@/components/Users/LinkedFacilities";
+import { UserModel } from "@/components/Users/models";
+
+type Props = {
+  userData: UserModel;
+  username: string;
+  refetchUserData?: () => void;
+};
```
Can't remove that as it's being passed as a prop to UserColumns function.

Jacobjeevan commented on src/components/Users/LinkedFacilities.tsx:
```diff
@@ -0,0 +1,303 @@
+  userData: UserModel;
+  refetchUserData?: () => void;
+}) {
+  const [facility, setFacility] = useState<any>(null);
```
Best to keep it that way as facilitySelect supports array of FacilityModel items, however we are restricting selection to just one in LinkedFacilities.

Jacobjeevan committed:
RabbitAI comments

Jacobjeevan committed:
Suggestions and error fixes

Jacobjeevan committed:
UsersManage tests

- Modified UsersManage tests
- Minor additions to components (Ids, names) to facilitate testing

Jacobjeevan commented on src/components/Users/UserListAndCard.tsx:
```diff
@@ -0,0 +1,374 @@
+    <>
+      <div className="mb-4 flex flex-col items-center justify-between gap-3 sm:flex-row">
+        <div className="sm:w-1/2">
+          <SearchInput
+            id="search-by-username"
+            name="username"
+            onChange={(e) => onSearch(e.value)}
+            value={searchValue}
+            placeholder={t("search_by_username")}
+          />
```
Already handled elsewhere (displays a no results banner if there aren't any results and searchbox is reset on clearing of the badge).

Jacobjeevan committed:
Fixed Usercreation tests

Jacobjeevan committed:
UsersHomepage tests, responsive design changes

- And tab functionality change (bringing in state change upwards to ensure tab selections stays through
search behavior)

Jacobjeevan committed:
separated out form components to diff functions

Jacobjeevan committed:
added username to more details btn Id for testing

Jacobjeevan commented on src/components/Users/UserAddEditForm.tsx:
```diff
@@ -0,0 +1,1232 @@
+            : formData.alt_phone_number === "+91"
+              ? ""
+              : formData.alt_phone_number,
+        ) ?? "",
+      date_of_birth: dateQueryString(formData.date_of_birth),
+      qualification:
+        formData.user_type === "Doctor" || formData.user_type == "Nurse"
+          ? formData.qualification
+          : undefined,
+      doctor_experience_commenced_on:
+        formData.user_type === "Doctor"
+          ? dayjs()
+              .subtract(
+                parseInt(formData.doctor_experience_commenced_on ?? "0"),
+                "years",
+              )
+              .format("YYYY-MM-DD")
+          : undefined,
+      doctor_medical_council_registration:
+        formData.user_type === "Doctor"
+          ? formData.doctor_medical_council_registration
+          : undefined,
+    };
+
+    const { res, error } = await request(routes.addUser, {
+      body: data,
+    });
+    if (res?.ok) {
+      dispatch({ type: "set_form", form: initForm });
+      Notification.Success({
+        msg: "User added successfully",
+      });
+      navigate("/users");
+    } else {
+      Notification.Error({
+        msg: error?.message ?? "Error while adding user",
+      });
+    }
+    setIsLoading(false);
+  };
```
You have already mentioned this in a previous comment.

Jacobjeevan committed:
Test fixes, more suggestions

Jacobjeevan commented on src/Utils/permissions.ts:
```diff
@@ -10,21 +31,30 @@ export const showUserDelete = (authUser: UserModel, targetUser: UserModel) => {
+  authUser: UserModel,
+  targetUser: UserModel,
+) => {
+  if (authUser.username === targetUser.username) return true;
+
+  // Auth user should be higher in hierarchy than target user
   if (
-    authUser.user_type === "DistrictAdmin" &&
-    targetUser.district_object?.id === authUser.district
+    USER_TYPES.indexOf(authUser.user_type) <=
+    USER_TYPES.indexOf(targetUser.user_type)
   )
-    return true;
+    return false;
 
-  return false;
+  return checkIfStateOrDistrictAdminInSameLocation(authUser, targetUser);
```
No, both of those objects should be present when showUserPasswordReset is called.

Jacobjeevan commented on cypress/pageobject/Users/ManageUserPage.ts:
```diff
@@ -68,6 +64,38 @@ export class ManageUserPage {
     cy.get(".error-text").should("contain", expectedError).and("be.visible");
   }
 
+  editUserDetails(
+    fName: string,
+    lName: string,
+    email: string,
+    dateOfBirth: string,
+    gender: string,
+  ) {
+    cy.get("#first_name").click().clear();
+    cy.get("#first_name").click().type(fName);
+    cy.get("#last_name").click().clear();
+    cy.get("#last_name").click().type(lName);
+    cy.get("#email").click().clear();
+    cy.get("#email").click().type(email);
+    cy.clickAndTypeDate("#date_of_birth", dateOfBirth);
+    cy.get("#gender").click();
+    cy.get("[role='option']").contains(gender).click();
+  }
+
+  verifyEditUserDetails(
+    fName: string,
+    lName: string,
+    email: string,
+    dateOfBirth: string,
+    gender: string,
+  ) {
+    cy.get("#first_name").should("have.value", fName);
+    cy.get("#last_name").should("have.value", lName);
+    cy.get("#email").should("have.value", email);
+    cy.get("#date_of_birth").should("have.value", dateOfBirth);
+    cy.get("#gender").should("contain.text", gender);
+  }
```
Unnecessary, as the method is called to check input provided earlier in the test.

Jacobjeevan commented on src/components/Users/UserListAndCard.tsx:
```diff
@@ -0,0 +1,413 @@
+      <div className="flex flex-row justify-between gap-x-3">
+        <div className="flex flex-col">
+          <div className="flex items-center gap-x-3">
+            <h1 id={`name-${user.username}`} className="text-base font-bold">
```
username is unique.

Jacobjeevan commented on src/components/Users/UserListAndCard.tsx:
```diff
@@ -0,0 +1,413 @@
+      return defaultAllowedUserTypes;
+  }
+};
+export const CanUserAccess = (user: UserModel | UserAssignedModel) => {
+  const allowedTypes = GetUserTypes(true).map((type) => type.id);
+  return allowedTypes.includes(user.user_type);
+};
```
User type is expected to be present for all users.

Jacobjeevan commented on cypress/pageobject/Users/ManageUserPage.ts:
```diff
@@ -116,10 +152,45 @@
+    cy.get("#user-edit-form").should("be.visible");
+  }
+
+  verifyDoctorQualification() {
+    cy.get("input[name='qualification']").should("be.visible");
+  }
+
+  verifyLinkedSkillsTabPage() {
+    cy.get("#select-skill").scrollIntoView();
+    cy.get("#select-skill").should("be.visible");
+  }
```
Not needed as the method is called specifically to verify the content for Doctor roles, however a similar method could be added to verify the element not occurring for other roles.

Jacobjeevan commented on cypress/e2e/users_spec/UsersManage.cy.ts:
```diff
@@ -85,13 +121,17 @@ describe("Manage User", () => {
     manageUserPage.clickSubmit();
-    manageUserPage.verifyErrorText("Value should be between 0 and 168");
-    // verify the data is reflected in user card and profile page
+    manageUserPage.verifyErrorText(
+      "Average weekly working hours must be a number between 0 and 168",
+    );
```
Not needed as testing is done with English.

Jacobjeevan committed:
minor additions

Jacobjeevan commented:
Cypress test 4 would only work once BE changes are merged (addition of new route).

Jacobjeevan committed:
switched to using userListview's get route

Jacobjeevan committed:
Added a profile view for read-only

- Render for users who won't have permissions to edit the profile
- Additional styling changes may be necessary
- Hide qualifications/skills and linked facilities tab for users without permissions

Jacobjeevan committed:
User View

- Additional changes to facilitate user view
	- Added edit profile button
	- Remove cancel button from the edit form (submit only)
- Removed UserQualifications
	- Instead using it in edit form
- Additional changes include adjusting the tab names (for skills tab), moving additional form fields in
add user form to their own functions

Jacobjeevan committed:
UserInformation

- Removed the edit button and replaced it with tabs
	- between edit and view for user information
- Minor styling changes
- Added a button to password reset form
	- Reveals/hides the form

nihal467 committed:
Merge branch 'develop' into issues/8878-user-management-redesign

Jacobjeevan committed:
UserDetails

- Separated out user details into 3 sections
	- Each with its own view/edit toggle button
- Modified UserAddEdit form to support this
	- Takes in fields as a prop
- Other minor changes to help with phasing out Profile component
	- Added language picker dropdown to user details page
	- Adjusted the sidebar notes based on if it's the user's personal page

Jacobjeevan committed:
UserSummary Tab Changes

- moving out the form/view components into UserSummary
- allowing for separate heading/note sections for each

Jacobjeevan committed:
Merge branch 'issues/8878-user-management-redesign' of https://github.com/Jacobjeevan/care_fe into issues/8878-user-management-redesign

nihal467 commented:
@Jacobjeevan Saw ongoing discussions on the User Management page design in Slack and the Cypress test failures. Once the PR is frozen, please ensure all issues with cypress are resolved and mark it as "Needs Testing."

Jacobjeevan committed:
Tests and minor fixes

- Ensure weekly working hours and video conference link is rendered for everyone
- Minor refactoring
	- Moving out toggle button as common element for user detail components
- Added and/or modified tests for user detail components

Jacobjeevan committed:
Merge branch 'develop' into issues/8878-user-management-redesign

Jacobjeevan commented on src/components/Users/UserFormValidations.tsx:
```diff
@@ -0,0 +1,148 @@
+  first_name: string;
+  last_name: string;
+  email: string;
+  phone_number: string;
+  alt_phone_number: string;
```
Additional phone number validation is being worked on in another issue.

Jacobjeevan commented on src/components/Users/UserFormValidations.tsx:
```diff
@@ -0,0 +1,148 @@
+  return null;
+};
+
+export const ValidateDoctorMedicalCouncilRegistration = (
+  formData: UserForm | EditForm,
+  translator: TFunction,
+) => {
+  if (
+    formData.user_type === "Doctor" &&
+    !formData["doctor_medical_council_registration"]
+  ) {
+    return translator("medical_council_registration_required");
+  }
+  return null;
+};
```
Is there a known format for this? Presumably we would want to accept registration numbers from around the world.

Jacobjeevan commented on src/components/Users/UserEditDetails.tsx:
```diff
@@ -0,0 +1,177 @@
+        t={t}
+      />
+      {isEditing ? (
+        <UserAddEditForm
+          username={username}
+          includedFields={editBasicInfoFields}
+          onSubmitSuccess={() => {
+            setIsEditing(false);
+            onSubmitSuccess?.();
+          }}
+        />
```
Already being handled in UserAddEditForm.

Jacobjeevan committed:
rabbit suggestions

Jacobjeevan committed:
Additional coderabbit suggestions

Jacobjeevan commented on src/components/Users/UserListAndCard.tsx:
```diff
@@ -0,0 +1,407 @@
+        {user.home_facility_object?.name || t("no_home_facility")}
+      </td>
+      {showDistrictColumn && (
+        <td id="district" className="px-4 py-4 text-sm">
+          {"district_object" in user && user.district_object
+            ? user.district_object?.name
+            : "district" in user && user.district
+              ? user.district
+              : ""}
+        </td>
```
No, as getDistrict is returning a JSX Element that's styled according to the card view.

Jacobjeevan commented on cypress/pageobject/Users/UserSearch.ts:
```diff
@@ -23,7 +23,7 @@ export class UserPage {
 
   checkUsernameText(username: string) {
-    cy.get(this.usernameText).should("contain.text", username);
+    cy.get(`${this.usernameText}-${username}`).should("contain.text", username);
```
No, this is checking for username in a different page.

Jacobjeevan committed:
more coderabbit suggestions

Jacobjeevan commented on src/components/Users/UserFormValidations.tsx:
```diff
@@ -0,0 +1,161 @@
+  return null;
+};
+
+export const ValidateVideoLink = (
+  formData: UserForm,
+  translator: TFunction,
+) => {
+  if (!formData["video_connect_link"]) return null;
+
+  try {
+    const parsed = new URL(formData["video_connect_link"]);
+    if (!["https:", "http:"].includes(parsed.protocol)) {
+      return translator("invalid_url_http_https");
+    }
+    if (parsed.href.toLowerCase().includes("javascript:")) {
+      return translator("invalid_url_javascript");
+    }
+  } catch {
+    return translator("invalid_url");
+  }
+  return null;
+};
```
No.

Jacobjeevan commented on src/components/Users/UserFormValidations.tsx:
```diff
@@ -0,0 +1,161 @@
+  return null;
+};
+
+export const ValidateDoctorExperienceCommencedOn = (
+  formData: UserForm,
+  translator: TFunction,
+) => {
+  if (formData.user_type === "Doctor") {
+    if (!formData["doctor_experience_commenced_on"]) {
+      return translator("doctor_experience_required");
+    } else if (
+      !validateNumber(formData["doctor_experience_commenced_on"] ?? "") ||
+      Number(formData["doctor_experience_commenced_on"]) < 0 ||
+      Number(formData["doctor_experience_commenced_on"]) > 100
+    ) {
+      return translator("doctor_experience_error");
+    }
+  }
+  return null;
+};
```
Not needed, what's there now is enough.

Jacobjeevan commented on src/components/Users/UserSummary.tsx:
```diff
@@ -0,0 +1,221 @@
+    return;
+  }
+
+  const handleSubmit = async () => {
+    setIsDeleting(true);
+    const { res, error } = await request(routes.deleteUser, {
+      pathParams: { username: userData.username },
+    });
+    setIsDeleting(false);
+    if (res?.status === 204) {
+      Notification.Success({
+        msg: t("user_deleted_successfully"),
+      });
+      setshowDeleteDialog(!showDeleteDialog);
+      navigate("/users");
+    } else {
+      Notification.Error({
+        msg: t("user_delete_error") + ": " + (error || ""),
+      });
+      setshowDeleteDialog(!showDeleteDialog);
+    }
+  };
```
If there's an error (most likely some sort of backend/network issue), notification is shown anyways. Closing the delete dialog so user doesn't immediately try to delete the user again.

Jacobjeevan committed:
added user delete tests

Jacobjeevan commented on cypress/pageobject/Users/ManageUserPage.ts:
```diff
@@ -1,10 +1,6 @@
-
-  clickFacilitiesTab() {
-    cy.get("#facilities").click();
+    cy.get("#home-facility").should("contain.text", expectedText);
```
don't worry about it.

Jacobjeevan committed:
fixes and suggestions

Jacobjeevan committed:
updated notes for user details view

Jacobjeevan commented:
Minor changes to the notes displayed (ex: "View or update user's personal information" -> "View user's personal information" if the user has no edit permissions).

Ready for review and testing.

Jacobjeevan committed:
changed password perms to reflect changes in BE

Jacobjeevan committed:
Merge branch 'develop' into issues/8878-user-management-redesign

nihal467 commented:
@Jacobjeevan can you clear the merge conflict, its a huge conflict

Jacobjeevan committed:
Merge branch 'develop' into issues/8878-user-management-redesign

Jacobjeevan commented:
> @Jacobjeevan can you clear the merge conflict, its a huge conflict

Done @nihal467

nihal467 commented:
**Test Progress**

**Question**

![image](https://github.com/user-attachments/assets/a7072c0b-180e-4d4b-8f7d-aadc7d0e6586)

- [ ] Only the account owner should be able to edit their own user information. However, with your current PR, an admin can edit other users' profiles. Is this the intended behavior?

nihal467 commented:
@Jacobjeevan check the test report, clear the conflict and attach a screenshot cypress passing in local

Jacobjeevan committed:
perms changes, form reset, sticky header on list

Jacobjeevan committed:
Merge branch 'develop' into issues/8878-user-management-redesign

Jacobjeevan commented:
> **Test Report**
> 
> **Functionality Bug**
* [x]  When you search for a username, the input field disappears. It only reappears after clearing the badge. Please revert this to the original behavior where the search field remains visible at all times.
* [x]  Similar issue noticeable for advanced filter as well
* [x]  Disable the "Add Facility" button for non-admin users on their own pages.
* [x]  Allow non-admin users to unlink their home facility and any facility in their access.
* [x]   The avatar on the user profile page and the profile icon are different. Please make them consistent.
* [x]  Reset the component's state after the password has been successfully changed.

> **Question For Clarity**
>     * [ ]  Only the account owner should be able to edit their own user information. However, with your current PR, an admin can edit other users' profiles. Is this the intended behavior?

Yep, that's the intention. 

>     * [ ]  The new profile view details page and the personal profile page display nearly identical data. Could we consider combining them into a single page for simplicity and consistency? [Discuss with the core team].

Yep, we are merging those (removing current profile page), I was waiting on it since there was an intern PR open updating elements in Profile page. We can open an issue that interns can take up once this is merged

* [x]  Ensure the first column remains fixed in the mobile view for the user list. This will improve accessibility and ease of navigation.

>     * [ ]  Add a "Last Modified" column in the user view page to display the date and time when the profile was last updated. Additionally, consider implementing an alert system to notify users to reverify their profile details after a period of 6 months or 1 year. This will help ensure that user information remains up-to-date. [Discussion required with core team before proceeding]

🤔  Yea could do for enhancement.

Here are the tests passing locally:

<img width="605" alt="image" src="https://github.com/user-attachments/assets/3592e7fd-be89-48b9-b934-fb059f4eecda">

Details on the error (screenshot below). I sometimes encounter this when testing after resetting the db; bypass it by manually logging in the cypress window.

![image](https://github.com/user-attachments/assets/a897b218-2cbd-48cd-b59a-f7a687029d19)

Jacobjeevan committed:
phone validation msg

rithviknishad commented on src/Utils/request/api.tsx:
```diff
@@ -1001,7 +1001,7 @@ const routes = {
 
   getUserDetails: {
-    path: "/api/v1/users/{username}/",
+    path: "/api/v1/users/{username}",
```
wouldn't this cause a http 301/302 redirect?

Jacobjeevan replied:
> wouldn't this cause a http 301/302 redirect?

🤔  Doesn't seem to be causing any issues, I'll add it though.

rithviknishad commented on src/common/validation.tsx:
```diff
@@ -57,6 +57,11 @@ export const validatePincode = (pincode: string) => {
   return pattern.test(pincode);
 };
 
+export const validateNumber = (number: string) => {
+  const pattern = /^[0-9]+$/;
+  return pattern.test(number);
+};
+
```
would this allow negative numbers?

also can we could use zod validators for newer forms.

Jacobjeevan replied:
> would this allow negative numbers?
> 
> also can we could use zod validators for newer forms.

Nah, only positive numbers

rithviknishad commented on src/components/Common/UserColumns.tsx:
```diff
@@ -0,0 +1,31 @@
+  refetchUserData?: () => void;
+};
+
+export default function userColumns(
```
We can make this a react component and args could be passed as props right?

```suggestion
export default function UserColumns(
```

Jacobjeevan replied:
Yep, can do.

nihal467 commented:
@Jacobjeevan the user management test are failing, fix it before requesting a review

rithviknishad commented on src/components/Common/UserDetails.tsx:
```diff
@@ -7,8 +7,8 @@ function UserDetails(props: {
     <div className="mt-2" id={props.id}>
-      <div className="font-light leading-relaxed text-secondary-900">
-        {props.title}:
+      <div className="text-sm font-light leading-relaxed text-[#B9B9B9]">
```
arbitrary colours are discouraged. let's stick to the design system. use one of gray shade color.

rithviknishad commented on src/components/Facility/FacilityUsers.tsx:
```diff
@@ -52,256 +27,44 @@ export default function FacilityUsers(props: any) {
+    routes.getFacilityUsers,
+    {
+      query: {
+        limit: resultsPerPage.toString(),
```
`request` used by useQuery will convert to string for you.

```suggestion
        limit: resultsPerPage,
```

nihal467 commented:
![image](https://github.com/user-attachments/assets/a95d11dc-6eec-4db1-b1a1-f132f360b567)

- [ ] the position of badges doesn't look good, we need to think of a different position , which can be taken as a separate issue

nihal467 commented:
conditonally approving it, @khavinshankar @rithviknishad merge only after relevant backend is merged and the cypress is passing

rithviknishad reviewed (commented):
We could've used shadcn's components and shadcn's form with zod validators in re-designs. But let's do that as separate PR since this is tested.

Jacobjeevan commented:
> ![image](https://private-user-images.githubusercontent.com/57055998/390605509-a95d11dc-6eec-4db1-b1a1-f132f360b567.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3OTMzMTksIm5iZiI6MTczMjc5MzAxOSwicGF0aCI6Ii81NzA1NTk5OC8zOTA2MDU1MDktYTk1ZDExZGMtNmVlYy00ZGIxLWIxYTEtZjEzMmYzNjBiNTY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI4VDExMjMzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhZmI2MWNiYTQxMzkxMjZkNTMxZWIyZDQ2NGZmMjViZjMzNGE5ZWFlMTVhYjI4YjI0ZjE2ZjYyMzliNTZhN2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.BCYf8sTXbPmLgctvZw9o3z2AWDcPeYC75yKz2Uu2ki4)
> 
>     * [ ]  the position of badges doesn't look good, we need to think of a different position , which can be taken as a separate issue

How's this?

<img width="1211" alt="image" src="https://github.com/user-attachments/assets/114b5687-e729-4d4d-a5a7-eaf7ea9c097a">

Jacobjeevan committed:
changes based on review comments

Jacobjeevan committed:
user create phone invalid messsage

Jacobjeevan committed:
Merge branch 'develop' into issues/8878-user-management-redesign

SwanandBhuskute committed:
fixing back button for camera asset

SwanandBhuskute committed:
Merge branch 'develop' into issues/9233/fixedbackButtonForCamera

Jacobjeevan committed:
Merge branch 'issues/9233/fixedbackButtonForCamera' of github.com:SwanandBhuskute/care_fe into issues/8878-user-management-redesign

Jacobjeevan committed:
Merge branch 'develop' into issues/8878-user-management-redesign

nihal467 commented:
@Jacobjeevan what is the status on this PR

Jacobjeevan commented:
> @Jacobjeevan what is the status on this PR

Still waiting on the backend 👍  I'll get the conflicts addressed though.

Jacobjeevan committed:
Merge branch 'develop' into issues/8878-user-management-redesign

Jacobjeevan committed:
Merge branch 'develop' into issues/8878-user-management-redesign

Jacobjeevan committed:
validateRule in UserAdd

Jacobjeevan committed:
Merge branch 'develop' into issues/8878-user-management-redesign

rithviknishad committed:
Merge remote-tracking branch 'origin' into issues/8878-user-management-redesign

Jacobjeevan committed:
removed superuser in password reset change perms

Jacobjeevan committed:
test fix
